arvo: unflop the spur in scry

Fang- opened the pull request:
As requested, no longer use reversed paths in sley-style peek handling.

I think this is the last of the "reverse path" shenanigans.

joemfb reviewed (changes requested):
This looks great!

While you're at it, could you do `+en-beam`/`+de-beam`? It doesn't look like too many call-sites even need to change.

Fang- commented:
@joemfb moving the `flop` out to callsites felt a bit sloppy to me, so I ended up straightening out the paths in everything related to their callsites. I need to test this a little bit more thoroughly, but everything seems fine so far.

(Only thing I didn't touch was the path format in the glass and metal pill generators, didn't feel quite comfortable enough with those.)

Snuck in a couple list utilities that I've been using in personal projects for a while now. They come in very handy for all these newly created "take/cut mark off path" cases. Hope this is ok!

Fang- commented:
idk what CI is up to, but tests are definitely failing. Looks like my fault for the most part, but not sure what this is about:

```
CRASHED /sys/hoon/mock/test-conditional-skips
/ted/test/hoon:<[68 12].[68 37]>
/ted/test/hoon:<[68 20].[68 36]>
/tests/sys/hoon/mock/hoon:<[6 3].[31 5]>
/tests/sys/hoon/mock/hoon:<[7 5].[9 38]>
/tests/sys/hoon/mock/hoon:<[9 7].[9 38]>
/tests/sys/hoon/mock/hoon:<[9 11].[9 38]>
```

joemfb commented:
This looks great so far!

You'll need to push a pill for CI to be meaningful (given sys/ changes)

joemfb commented:
When you push a pill, you should also target `na-release/candidate` (and rebase/merge first to be up-to-date). Sorry for the churn!

Fang- commented:
> You'll need to push a pill for CI

Every single time.

Added in some additional fixes for the un/mount commands, made octs auraless again, and updated the pill. Running tests locally, still getting the `test-conditional-skips` failure that I reported earlier, but I'm 99% sure that's unrelated, and everything else should be fine.

Tagging @philipcmonk to do a sanity check on the clay changes here. I _think_ this was the last bit of its interface that still used old-style/reverse spurs?

joemfb commented:
I'm kind of shocked that's the only problem you saw using the wrong binary.

joemfb commented on pkg/arvo/sys/zuse.hoon:
```diff
@@ -76,7 +76,7 @@
 +$  rift  @ud                                           ::  ship continuity
 +$  mime  (pair mite octs)                              ::  mimetyped data
-+$  octs  (pair @ud cord)                               ::  octet-stream
++$  octs  (pair @ud @)                                  ::  octet-stream
```
My bad -- good catch!

joemfb reviewed (approved):
This all LGTM, TYFYS!

philipcmonk reviewed (approved):
There's probably a few reversed paths in generator comments, but that can be cleaned later.  Or left as a warning to those who come after!
